### PR TITLE
Visual polish for 0.5.0

### PR DESCRIPTION
**Pull request by hobinjk**

### CODE REVIEW

**Comment by benfrancis** [comment]
Thanks for doing this. I had generally avoided drop shadows in order to keep a flat visual design, but this is very subtle and does solve the problem on the floorplan, so nice idea.

Sorry if my screenshots in #1198 confused matters, but my intention wasn't to completely remove the padding on the things screen, just to make it part of the scrollable area rather than outside it. i.e. in the initial state it should look like this:
screen shot 2018-07-30 at 16 00 31

rather than this:
screen shot 2018-07-30 at 16 00 32

...but should then scroll underneath the buttons (as it was in 0.4) rather than disappearing behind an invisible border (as it currently is in master). We can then potentially also add a Things title which scrolls off the page, as per #1211.

One other thing, I've noticed a regression in the rules screen where the menu button now appears instead of the back button:
screen shot 2018-07-30 at 15 54 46

**Comment by hobinjk** [comment]
Right, makes sense. I've fixed most of the issues but noticed an overarching problem with the approach I took to calling out the buttons which also applies to the use of webcomponents on the floorplan. The issue is that the icons have a low opacity fill. This blends well into any backdrop but also blends into the icons on the things list. I fixed this by giving them the light background color to make them stand out from the things list. However, this backfires on the floorplan's edit view where the backdrop is dark instead of light so the manual blending doesn't look correct. The webcomponents also have an incorrect background based on the light one.

screen shot 2018-07-30 at 11 49 54

The webcomponents are straightforward to fix but this means that I'll have to take a different approach on the buttons if we want them to be visually distinct. The easiest way to fix this is to do what the thing edit view does. It hides all irrelevant buttons on the floorplan edit view and specifies its own back button which can then be styled correctly. This would also fix an issue where you can navigate away in the middle of editing the floorplan and leave the screen in a weird intermediate state.

tl;dr: spotted overarching issue while fixing minor issues, will fix and update PR accordingly

**Comment by hobinjk** [comment]
Hmm, the icon opacity is actually specified in the css, not the svg. It doesn't change the approach I'll be taking but I'm updating here so there are no misconceptions left by my previous comment.

**Comment by hobinjk** [comment]
screen shot 2018-07-30 at 12 50 00

With the buttons now more muted I'm more okay with the components sticking out like that, let me know if you want me to change anything.

**Comment by hobinjk** [comment]
Hmm, sorry for forgetting about Settings. Looking into it the design of the view, it would be a major refactoring to get it to scroll properly. It would be a very nice improvement though so if I get any time I'll try to tackle it.
